src/shared/mount-util.c

```diff
@@ -77,11 +77,34 @@ int umount_recursive(const char *prefix, int flags) {
         return n;
 }
 
-static int get_mount_flags(const char *path, unsigned long *flags) {
-        struct statvfs buf;
+/* Get the mount flags for the mountpoint at "path" from "table" */
+static int get_mount_flags(const char *path, unsigned long *flags, struct libmnt_table *table) {
+        struct statvfs buf = {0};
+        struct libmnt_fs *fs = NULL;
+        const char *opts = NULL;
+        int r = 0;
+
+        fs = mnt_table_find_target(table, path, MNT_ITER_FORWARD);
+        if (fs == NULL) {
+                log_warning("Could not find '%s' in mount table", path);
+                goto fallback;
+        }
+
+        opts = mnt_fs_get_vfs_options(fs);
+        r = mnt_optstr_get_flags(opts, flags, mnt_get_builtin_optmap(MNT_LINUX_MAP));
+        if (r == 0) {
+                log_warning_errno(r, "Could not get flags for '%s': %m", path);
```

Comment:
hmm, isn't mnt_opstr_get_flags() returning 0 success? i am confused...

Reply:
Oh! Yes it does, and this indeed prints lots of "Could not get flags for...: Success" into the log on bootup. I'm sorry that I missed that in my testing. Fixed now, check is inverted, messages are gone.